supportedBackends/google/batch/src/main/scala/cromwell/backend/google/batch/models/GcpBatchExitCode.scala

```diff
@@ -0,0 +1,41 @@
+package cromwell.backend.google.batch.models
+
+sealed abstract class GcpBatchExitCode(val code: Int) extends Product with Serializable
+
+/**
+ * Represents the possible exit codes from Batch.
+ *
+ * See: https://cloud.google.com/batch/docs/troubleshooting#reserved-exit-codes
+ */
+object GcpBatchExitCode {
+  case object VMPreemption extends GcpBatchExitCode(50001)
+
+  case object VMReportingTimeout extends GcpBatchExitCode(50002)
+
+  case object VMRebootedDuringExecution extends GcpBatchExitCode(50003)
+
+  case object VMAndTaskAreUnresponsive extends GcpBatchExitCode(50004)
+
+  case object TaskRunsOverMaximumRuntime extends GcpBatchExitCode(50005)
+
+  case object VMRecreatedDuringExecution extends GcpBatchExitCode(50006)
+
+  val values: List[GcpBatchExitCode] = List(
+    VMPreemption,
+    VMReportingTimeout,
+    VMRebootedDuringExecution,
+    VMAndTaskAreUnresponsive,
+    TaskRunsOverMaximumRuntime,
+    VMRecreatedDuringExecution
+  )
+
+  // Special case non-error code - this does not correspond to any messages we get from Batch,
+  // this is what we use for tasks that don't error out at the Batch layer.
+  case object Success extends GcpBatchExitCode(0)
+
+  def fromEventMessage(message: String): Option[GcpBatchExitCode] =
+    values.find { target =>
+      message.contains(s"exit code ${target.code}")
```

Comment:
Any chance the pattern could be made more specific? "exit code 0" seems like something that might appear in a message without necessarily being of Batch origin.

Reply:
Definitely understand the concern about `exit code 0`, but the special `Success` code isn't included in the list of values we're checking against here - it doesn't correspond to a real status we get from GCP and can't be returned from this method. Should add a comment calling that out. Does that alleviate your worries (or create more worries)?

Reply:
Oh whoops that `values` fooled me...  I thought this Java `enum`-like construction had a literal `values()` method like the one Java enums have, returning an array of **all** instances in the order they were defined. I agree a comment (and maybe a renaming) would be helpful.

`exit code 0` was the only false positive I was really worried about, so if that isn't among the list of values searched I'm not worried.